src/components/article/ViewArticle.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { getArticle } from '../../actions/articles';
+import NavBar from '../presentation/NavBar';
+import DisplayMessage from '../presentation/DisplayMessage';
+import './ViewArticle.scss';
+import Footer from '../presentation/Footer';
+
+class ViewArticle extends React.Component {
+  state={
+    loading: 'https://i.imgur.com/ungt2Pg.gif',
+  };
+
+  componentDidMount = async () => {
+    const { match: { params }, getArticle: viewArticle } = this.props;
+    await viewArticle(params.slug);
+    this.setState({ loading: '' });
+  }
+
+  render() {
+    const { loading } = this.state;
+    const {
+      tags,
+      message,
+      article,
+      author,
+    } = this.props;
+
+    const {
+      title,
+      body,
+      imgUrl,
+      readtime,
+      createdAt,
+    } = article;
+    const { username } = author;
+    const date = new Date(createdAt).toDateString();
+    return (
+      <div>
+        <NavBar />
```

Comment:
Nice work applying your learnings here @oluwajuwon.
I noticed you added the `NavBar` component directly as a child of this component (`ViewArticle`). I believe the right place to put this is in the `App.jsx` file. The `NavBar` is an _always visible component_ that we don't want to always manually add every time.

Reply:
@codeshifu I left it that way because not all pages need the nav like the `login` and `sign up` pages, they won't have the navbar, but if I move the component to the `App.jsx` the navbar will show on all pages regardless.